target_iceberg/utils.py

```diff
@@ -43,7 +41,10 @@ def _field_type_to_pyarrow_field(
     pyarrow_type = FIELD_TYPE_TO_PYARROW.get(input_type, pa.string())
     # override with timestamp type if format set to date or date-time even if value type is e.g. string.
     if input_types.get("format") in ["date", "date-time"]:
-        pyarrow_type = pa.timestamp("us")
+        pyarrow_type = pa.timestamp("us", tz="UTC")
+    # ensure all timestamps are UTC and not converted e.g. to timestamp_ntz
+    if pa.types.is_timestamp(pyarrow_type):
+        pyarrow_type = pa.timestamp("us", tz="UTC")
```

Comment:
If we’re already defining it using property/format with the correct PyArrow type, does this step make any difference? I don’t see any case where we need to explicitly ensure it.

Reply:
Good question, empirically it would seem it does make a difference, as the scratch table was created with different types. I think this might be useful if we get timestamps from the tap, rather than a string, which we convert ourselves. Isn't that possible for some taps?